### ISSUE TITLE
Getting indices of a sparse compressed tensor results tensors with requires_grad being True 

### ISSUE THREAD

**Issue by pearu** [pytorch/pytorch#107083]
## Issue description

As in the title.

It is a bug because "only Tensors of floating point dtype can require gradients".

## Code example

```python
>>> a = torch.tensor([[1, 2], [3, 4]], dtype=torch.float64, requires_grad=True).to_sparse_csr()
>>> a
tensor(crow_indices=tensor([0, 2, 4]),
       col_indices=tensor([0, 1, 0, 1]),
       values=tensor([1., 2., 3., 4.]), size=(2, 2), nnz=4,
       dtype=torch.float64, layout=torch.sparse_csr,
       grad_fn=<ToSparseCsrBackward0>)
>>> a.crow_indices()
tensor([0, 2, 4], requires_grad=True)
```

Expected `a.crow_indices().requires_grad == False`.

## System Info

- PyTorch version: main

cc @alexsamardzic @nikitaved @cpuhrsch @amjames @bhosmer
